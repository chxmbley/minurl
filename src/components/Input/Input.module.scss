.input {
  @apply border border-slate-200 dark:border-slate-700 px-6 h-12 rounded-full shadow-lg text-lg dark:bg-slate-800 dark:text-slate-100 outline-none transition-colors;

  &focus {
    @apply border-cyan-500;
  }

  &:invalid {
    @apply text-red-600 dark:text-red-400 border-red-600 dark:border-red-600;
  }

  &:disabled {
    @apply cursor-not-allowed bg-slate-200 dark:bg-slate-700 text-slate-400 dark:text-slate-600;
  }
}
