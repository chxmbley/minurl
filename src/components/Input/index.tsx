import cn from 'classnames';
import { forwardRef, HTMLProps } from 'react';
import type { ForwardRefRenderFunction } from 'react';
import styles from './Input.module.scss';

/** Styled input component */
const Input: ForwardRefRenderFunction<HTMLInputElement, HTMLProps<HTMLInputElement>> = (
  { className, ...props },
  ref,
) => <input ref={ref} className={cn(className, styles.input)} {...props} />;

export default forwardRef(Input);
