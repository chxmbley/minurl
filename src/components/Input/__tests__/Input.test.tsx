import { render, screen } from '@testing-library/react';
import Input from '~components/Input';

test('renders an Input component', () => {
  const { container } = render(<Input />);
  expect(container).toMatchSnapshot();
});

test('adds a provided className to the component', () => {
  const customClassName = 'my-custom-classname';
  const { container } = render(<Input data-testid="input" className={customClassName} />);
  expect(container).toMatchSnapshot();
  expect(screen.getByTestId('input')).toHaveClass(customClassName);
});
