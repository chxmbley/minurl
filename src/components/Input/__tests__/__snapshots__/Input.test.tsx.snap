// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a provided className to the component 1`] = `
<div>
  <input
    class="my-custom-classname input"
    data-testid="input"
  />
</div>
`;

exports[`renders an Input component 1`] = `
<div>
  <input
    class="input"
  />
</div>
`;
